### DISCUSSION TITLE
How to filter iframes to execute the code

### DISCUSSION THREAD

**Discussion by Murka007** [Tampermonkey/tampermonkey#1582] [Q&A] [answered]
I want my tampermonkey script to be executed in specific iframes that contains `class` and `id` attributes. I don't want it to run in all iframes. How do I achieve this?
```html
// Should be executed in these iframes
<iframe class="page-frame" id="iframe-0"></iframe>
<iframe class="page-frame" id="iframe-1"></iframe>

// Should ignore this
<iframe class="random-frame" id="randomframe-0"></iframe>
```

**Comment by derjanb** [comment, marked as answer]
This is not possible. Scripts are only filtered by tab/frame URL. 
In case the iframe is embedded inside a page with the same origin you can manually filter where the script should finally do something.

**Reply by Murka007** [reply]
Yes, my iframes have the same origin. I could do something like this `window.top === window.self`
